### ISSUE TITLE
[BUG] QuestHandler fails to load after load after server restart

### ISSUE THREAD

**Issue by Chetral** [evennia/evennia#3560]
#### Describe the bug
After importing the contrib.tutorials.evadventure.quest module, setting up a simple quest and adding it to the character quest handler (lazy property, as described in the docs), the quest is assigned and saved properly. But if there is a server stop and start, the questhandler fails to load, giving a recursion error.

#### To Reproduce
1. in your Character.py:
 import contrib.tutorial.evadventure.quest 
`from evennia.contrib.tutorials.evadventure.quest import EvAdventureQuestHandler`
add lazy_property for questhandler:
```py
    @lazy_property
    def quests(self):
        return EvAdventureQuestHandler(self)
```
2. create a simple quest:
```py
from typeclasses.quests import Quest
from server.utils import move_coin

class LostItem(Quest):

    key = "LostItem"
    desc = "the lost ring"
    
    help_start = "find the lost ring"
    help_end = "give the ring to the owner"
    
    def step_start(self, *args, **kwargs):
        obj = [x for x in self.quester.contents if x.tags.has("LostItem", category="quest")]
        if len(obj) > 0:
            self.quester.msg("You have the ring. Take it back to the owner!")
            self.current_step = "end"
            self.progress()

    def step_end(self, *args, **kwargs):
        if self.quester.location == self.client.location:
            obj = [x for x in self.client.contents if x.tags.has("LostItem", category="quest")]
            if len(obj) > 0:
                obj[0].delete()
                self.quester.msg(f"{self.client.name.capitalize()} thanks a lot, peace and love!.")           
                self.complete()
```
4. add the quest to your character, I have done it trought a mob responding to actions, the code is:
```py
                        caller.quests.add(LostItem)
                        q = caller.quests.get('LostItem')
                        q.add_data('client', self)
```
The last line adds the attribute "client" to the quest, with value the mob itself. 
If you use default CmdQuest (from the contrib) you see the quest correctly assigned to you, in status "started". You can also access it and see the attribute "client" is set up correctly.

5. restart the server

6. Using the standard CmdQuest you got the error:
![image](https://github.com/evennia/evennia/assets/55252937/6988cd9a-88b9-4d1d-bc06-ee23deab3684)

If you DO NOT use the command, but try to access the questhandler by command lines (py me.quests.all()), you will find the questhandler is empty, even if in your character there are the saved attributes

#### Expected behavior
At reload all quests should be reloaded with their status and optional data

#### Environment, Evennia version, OS etc
Evennia : 4.1.1 (rev 7f41e5a64)
OS: posix (ubuntu 24)
Python 3.11.3
Twisted 23.10.0
Django 4.2.11

**Comment by Griatch** [comment]
Noting that this needs to be addressed as I continue working on the beginner tutorial.

**Comment by Griatch** [comment]
Indeed, there are some overall lack of polish and testing for the quest tutorial overall. I have added a warning to the lesson to note that it's not ready for use straight up at this time. As I wrote above, I need to get back to it.
